docs/sources/developers/plugins/extend-the-grafana-ui-with-links.md

```diff
@@ -1,25 +1,78 @@
 ---
-title: Use extensions to add links to app plugins
+title: Use extensions to add new functionality in the Grafana user interface
 description: Learn how to add links to the Grafana user interface from an app plugin
 ---
 
-# Use extensions to add links to app plugins
+# The when, why and how of UI extensions
 
-You can use the Plugin extensions API with your Grafana app plugins to add links to the Grafana UI. This feature lets you send users to your plugin's pages from other spots in the Grafana application.
+Before we go into details we need to cover the two major concepts in the UI extensions feature of Grafana.
+
+1. Extension point - a place in the UI where plugins can provide new functionality to the end user. Examples of extension points are the dashboard panel menu or toolbar in explore.
+2. Extensions - New functionality, usually registered by an app plugin, that will be displayed at an extension point. Examples of extensions are the possibility to create an incident directly from the dashboard panel menu.
+
+![panel menu](https://user-images.githubusercontent.com/172951/242723354-a10d6238-22f1-4458-b85e-ac3c7f014b22.png)
+
+In the example above we have one extension point with two extensions registered. This highlights one of the benefits of using UI extensions. Once you have added an extension point to your UI it can be extended multiple times by multiple plugins.
+
+## When should I add an extension point?
+
+If you have parts of your UI where it would improve the user experience to allow users to trigger functionality directly from the current view instead of navigating to the plugin. If this is the case, then it might be a good idea to consider adding an extension point.
+
+Examples where it would be useful:
+
+- The user views a dashboard with historical data. By adding an extension point to this part of the UI the Machine Learning app plugin can give the user the ability to create a forecast for that data directly from the panel.
+- The user views a firing alert. By adding an extension point to this part of the UI the Incident app plugin can give the user the ability to create an incident directly from the alert view.
+
+## Why should I add an extension point?
+
+Adding an extension point to your UI gives a number of benefits such as:
+
+- Add the UI extension point once to enable multiple plugins to extend your UI with new functionality. No additional effort needed to provide functionality from more plugins in your UI.
+- Clean separation of concerns. Your application does not need to know anything about the plugin extending your UI.
+- Integration build for change. Since your application doesn't know anything about the internals of the plugin extending your UI they are free to change their APIs without risk of breaking the extension point UI.
+- Easy to bootstrap. If both apps are installed and enabled the extensions will automatically be configured and displayed to the user.
```

Comment:
I think we need to expand on this section. I _believe_ you're referring to the removal of any requirement to query for available extensions or ascertain their availability/health status before deciding on options to present to a user?

Reply:
Being easy to add generally could be more emphasised, particularly in scenarios where extension points could be requested by those who wish to create an extension.